x/poolmanager/client/cli/query.go

```diff
@@ -68,6 +69,15 @@ func GetCmdNumPools() (*osmocli.QueryDescriptor, *queryproto.NumPoolsRequest) {
 	}, &queryproto.NumPoolsRequest{}
 }
 
+// GetCmdAllPools return all pools available across Osmosis modules.
+func GetCmdAllPools() (*osmocli.QueryDescriptor, *queryproto.AllPoolsRequest) {
+	return &osmocli.QueryDescriptor{
+		Use:   "all-pools",
+		Short: "Query all pools on the Osmosis chain",
+		Long:  "{{.Short}}",
+	}, &queryproto.AllPoolsRequest{}
+}
+
```

Comment:
Just a note here, under QueryDescriptor I think we need to add hasPagination to true. This will add the corresponding flags and such, but I don't think its properly paginates out of the box (there is no osmocli query that currently uses it, and any paginated query we actually rely on uses to old cli setup).

Reply:
Pagination isn't implemented for this query. Since they are they only known client, spoke to Jon and decided not to added for now due to time constraints